app/screens/custom_status/custom_status.tsx

```diff
@@ -243,6 +244,27 @@ const CustomStatus = ({
             return;
         }
 
+        const reachable = (await NetInfo.fetch()).isInternetReachable;
```

Comment:
I wonder if this behaves well with airgapped or VPN situations.

Why the handling done in 289 (on `await updateCustomStatus(serverUrl, status)`) is not workng?

Reply:
the issue reported was that the modal was getting blocked without a way to close it, and by seeing the code in that line 289 it makes total sense. The return happens before closing the window and the emmited event show the error message, but not in the modal and since there was not way to close the modal. Maybe, we can just rely in the current line 289 solution if you think that the proposed solution could give problem in airgapped or vpn, but in that error scenario besides returning also closing the modal. Thoughts? @larkox

Reply:
or do you think there is a better way to verify the server connection for this scenario?

Reply:
I will defer to @matthewbirtch , but I think we can "close always". Worst case scenario, the user lose a bit of information (the custom status they were trying to set) but I feel it is not a big deal.

Also (correct me if I am wrong) the kind of errors that this should show are mainly "you could not connect to the server" or "something that should never happen in the server happened" (like not finding the user, or stuff like that). So it is rare enough for this to be a good compromise.

Reply:
@matthewbirtch any thoughts on @larkox comments?

Reply:
Sorry @pvev, did my comment go through?

> Not really sure what line 289 does. From my testing, what I'm seeing is an alert dialog that shows when I close the custom status modal stating "Status couldn't be updated" (even if I haven't made a change). Then we show the error message inline with the custom status menu item. Personally, I don't think we need the alert. I think we should be fine with the Failed to update status. Try again error message that appears inline.

Reply:
yes, it did @matthewbirtch , somehow I missed it. Probably an opened tab that was not correctly updated. Check my last video posted in comments to see how its now working.